tests/dtypes/test_dataset_to_numpy.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import reciprocalspaceship as rs
+
+
+@pytest.mark.parametrize("dtype", [None, np.int32, np.float32, np.float64, object])
+@pytest.mark.parametrize("hasna", [True, False])
+def test_to_numpy_ints(dtype, hasna):
+    """
+    Test DataSet.to_numpy() with int32-backed MTZDtypes
+    """
+    ds = rs.DataSet(
+        {
+            "X": np.random.randint(0, 100, size=100),
+            "Y": np.random.randint(0, 100, size=100),
+        }
+    ).infer_mtz_dtypes()
```

Comment:
I for one find `infer_mtz_dtypes` to be a useful function, but I don't think you should rely on it for the purposes of this test. If the behavior changes in the future, it will make for some annoying debugging. Explicitly set these dtypes (likewise for the other tests).

Reply:
Fair point -- these tests should be minimally dependent on other features. I've updated the relevant clauses to explicitly set dtypes